src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -178,23 +179,35 @@ public void testProcessNonJavaFiles() throws Exception {
         treeWalker.configure(new DefaultConfiguration("default config"));
         final DefaultConfiguration childConfig = createCheckConfig(JavadocParagraphCheck.class);
         treeWalker.setupChild(childConfig);
-
+        final File file = new File("input.java");
         final List<String> lines =
             new ArrayList<>(Arrays.asList("package com.puppycrawl.tools.checkstyle;", "",
                 "error public class InputTreeWalkerFileWithViolation {}"));
-
+        final FileText fileText = new FileText(file, lines);
         try {
-            treeWalker.processFiltered(new File("input.java"), lines);
+            treeWalker.processFiltered(file, fileText);
             fail("Exception expected");
         }
         catch (CheckstyleException ex) {
             assertEquals("Invalid exception message",
                 "MismatchedTokenException occurred during the analysis of file input.java.",
                 ex.getMessage());
         }
+    }
 
-        // Should not throw exception as file with txt extension should not be processed at all
-        treeWalker.processFiltered(new File("input.txt"), lines);
+    @Test
+    public void testProcessNonJavaFilesWithoutException() throws Exception {
+        final TreeWalker treeWalker = new TreeWalker();
+        treeWalker.setTabWidth(1);
+        treeWalker.configure(new DefaultConfiguration("default config"));
+        final File file = new File("src/main/resources/checkstyle_packages.xml");
+        final FileText fileText = new FileText(file, StandardCharsets.ISO_8859_1.name());
+        try {
+            treeWalker.processFiltered(file, fileText);
+        }
+        catch (CheckstyleException ex) {
+            fail("Exception not expected");
```

Comment:
Like I mentioned to others, `try-catch-fail` is not a good option.
JUnit automatically populates exception to user and fails the test. If exception occurs, user will now get a bland message and not the stack trace of the problem.
There is nothing we can assert here?